device-connectors/src/testflinger_device_connectors/devices/zapper_iot/parser.py

```diff
@@ -0,0 +1,198 @@
+                                "required": ["method"],
+                            },
+                            "reboot_install": {
+                                "type": "object",
+                                "properties": {
+                                    "method": {"$ref": "#/$defs/method"},
+                                    "timeout": {
+                                        "type": "integer",
+                                        "default": 600,
+                                    },
+                                },
+                                "required": ["method"],
+                            },
+                            "sys_commands": {
+                                "type": "array",
+                                "items": {"type": "string"},
+                            },
+                            "eof_commands": {
+                                "type": "array",
+                                "minItems": 1,
+                                "items": {
+                                    "type": "object",
+                                    "properties": {
+                                        "cmd": {"type": "string"},
+                                        "expected": {"type": "string"},
+                                    },
+                                },
+                            },
+                        },
+                    },
+                ]
+            },
+        },
+        "period": {
+            "type": "object",
+            "properties": {
+                "mode": {"$ref": "#/$defs/mode"},
+                "day": {"$ref": "#/$defs/day"},
+                "time": {"$ref": "#/$defs/time"},
+            },
+            "required": ["mode"],
+            "allOf": [
+                {
+                    "if": {"properties": {"mode": {"enum": ["day"]}}},
+                    "then": {"required": ["time"]},
+                },
+                {
+                    "if": {"properties": {"mode": {"enum": ["week"]}}},
+                    "then": {"required": ["time", "day"]},
+                },
+            ],
+        },
+    },
+    "required": ["config", "run_stage"],
+    "$defs": {
+        "time": {"type": "string", "pattern": "^[0-2][0-9]:[0-5][0-9]$"},
+        "day": {
+            "type": "string",
+            "enum": ["mon", "tue", "wed", "thu", "fri", "sat", "sun"],
+        },
+        "mode": {"type": "string", "enum": ["hour", "day", "week", "test"]},
+        "mail_format": {
+            "type": "string",
+            "pattern": "^[a-zA-Z0-9.]+@[a-zA-Z0-9.]+$",
+        },
+        "method": {
+            "type": "string",
+            "enum": [
+                "cloud-init",
+                "console-conf",
+                "system-user",
+            ],
+        },
+    },
+}
+
+
+def validate_data(data):
+    """for verify provision yaml"""
+    try:
+        validate(instance=data, schema=LAUNCHER_SCHEMA)
```

Comment:
Since this seems to be looking at the "tplan" key from the data that we were passed, it seems to me like it might help if we:
1. name the function validate_tplan
2. rename LAUNCHER_SCHEMA to TPLAN_SCHEMA

I think it's ok to have this one schema here if that's the only one, but if we want to add others, then I would recommend at least putting them in a schemas module, and importing them from there. That way you can just focus on the code here.